### DISCUSSION TITLE
MD5 or Hashbytes function for SqlServer

### DISCUSSION THREAD

**Discussion by sdmimaye** [linq2db/linq2db#4319] [Q&A] [answered]
Hi there!

I would like to implement a function to calculate a MD5 hash. The function is currently only supported on SqlServer. My current approach:

    public static class DatabaseFunctions
    {
        [Sql.Expression("HASHBYTES('{0}', {1})", PreferServerSide = true)]
        public static byte[] Hashbytes(string hash, string content) => throw new NotImplementedException();

        [Sql.Function("HASHBYTES('MD5',{0})", PreferServerSide = true)]
        public static byte[] MD5(string text) => throw new NotImplementedException();
    }

Neither of those two are working. I followed the example on this site:
http://blog.linq2db.com/2016/06/how-to-teach-linq-to-db-convert-custom.html

Could someone point me to the right direction? 

Best regards
Michael

**Comment by MaceWindu** [comment, marked as answer]
- `Sql.Function` takes only function name and generate parameters based on attribute properties and list of C# function parameters.
- `Sql.Expression` takes full SQL fragment with placeholders

So in your case it should look like:

```cs
public static class DatabaseFunctions
{
    [Sql.Function("HASHBYTES", PreferServerSide = true)]
    public static byte[] Hashbytes(string hash, string content) => throw new NotImplementedException();

    [Sql.Expression("HASHBYTES('MD5',{0})", PreferServerSide = true)]
    public static byte[] MD5(string text) => throw new NotImplementedException();
}
```

Ping me if it doesn't work for you and I will look into it.

**Reply by sdmimaye** [reply]
Worked like a charm. Thank you very much!
